### PR TITLE
TECH-1850 - Migrating chief-keeper to Kubernetes

### PR DESCRIPTION
**Pull request by OleksandrUA**
TECH-1850 - Migrating chief-keeper to Kubernetes

### CODE REVIEW

File: .github/workflows/build-and-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+on:
+  push:
+    branches:
+      - master
+      - TECH-1850
```
Comment (sanbotto) [inline .github/workflows/build-and-deploy.yaml:5]:
```suggestion
```

**Review by jeannettemcd — COMMENTED** [review]
Chief Keeper has an EFS volume. I'm not seeing it, is it defined elsewhere ?

**Comment by OleksandrUA** [comment]
@jeannettemcd oh, right, thanks for noticing. I'll add it now

File: .github/workflows/build-and-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+name: Build Docker image and deploy it to Kubernetes cluster
+
+jobs:
+  deploy:
+    name: Build-Deploy
+    runs-on: ubuntu-latest
+    environment: production
+    env:
+      AWS_REGION: us-east-1
+      CLUSTER_NAME: maker-prod
```
Comment (eskp) [inline .github/workflows/build-and-deploy.yaml:16]: Do you think it is a security risk to expose deployment region and cluster name in a public repo?

**Comment by eskp** [comment]
@OleksandrUA we need to properly monitor these services before we can call migration complete. chief-keeper was failing on K8s due to not being able to connect to RPC endpoint, we didn't get any alerts and the keeper missed its duties. 

We need to 1) Monitor for pod crashes/restarts with Grafana/Prometheus 2) Create Loki queries to match strings with ERROR (or alike) and send alerts when threshold is crossed for these queries

**Comment by OleksandrUA** [comment]
> Chief Keeper has an EFS volume. I'm not seeing it, is it defined elsewhere ?

@jeannettemcd done, added both support of AWS EFS (TECH-1902) and actually the volume

**Review by eskp — APPROVED** [review]

**Review by sanbotto — APPROVED** [review]
I'm not quite sure about leaving the fileSystemId in plain text, though.

**Comment by cristidas** [comment]
@OleksandrUA the latest GHA build failed https://github.com/makerdao/chief-keeper/actions/runs/5622477750/job/15235211451

**Comment by OleksandrUA** [comment]
@cristidas yes, this is expected as build tries to use Helm Chart version `0.0.14` which is not yet published due to https://github.com/MakerOps/helm-charts/pull/16
That PR os approved and tested so I am going to merge that one first, wait for `0.0.14` gto be available and restart current PR with empty commit.

**Pull request merged**